### ISSUE TITLE
Error thrown when a subscription calls 'send' immediately

### ISSUE THREAD

**Issue by mattlenz** [yoshuawuyts/choo#29]
Let's say I'd like to trigger an action that initialises some things first up. If I have a model like so;

```js
app.model({
  state: {
    init: false
  },
  subscriptions: [
    send => send('init')
  ],
  reducers: {
    init: state => ({ ...state, init: true })
  }
})
```

The browser throws `TypeError: Cannot read property 'nodeType' of null`

I assume due to the view not being ready when the `send` occurs.

Changing the subscription function to `setImmediate(() => send('init'))` solves the issue, but, should this be handled by choo? Or is using subscriptions in this way not advised?

**Comment by yoshuawuyts** [comment]
Hey @mattlenz, thanks for opening this issue. This ties into the issue raised in https://github.com/yoshuawuyts/choo/pull/26; functions in `choo` can be called before the DOM tree has been applied to the `dom`; which means that a re-render is triggered on a node that doesn't exist.

I'd suggest instead of timeout though, to use:
```js
document.addEventListener('DOMContentLoaded', (e) => send('init'))
```
Minor difference, but should be more robust.

I think the resolution here is to `assert()` if the DOM tree exists for the update, and throw if it doesn't. This behavior should probably also be documented somewhere. That way the bug in the application is made explicit, and steps can be taken to fix it.

I hope this makes sense! I'll put through a PR to fix this bug. Cheers!

**Comment by yoshuawuyts** [comment]
Fix released as `v2.1.4` - let me know if this works for you. Thanks! :sparkles:

**Comment by mattlenz** [comment]
Amazing, thanks @yoshuawuyts!
